rclpy/rclpy/names_and_types.py

```diff
@@ -29,9 +29,8 @@ def __init__(self, **kwargs):
         self.topic_names = kwargs.get('topic_names', list())
         self.type_names = kwargs.get('type_names', list())
 
-    def __repr__(self):
-        s = 'number of topics = ' + repr(self._topic_count) + '\n'
-        s += '[topic_name] : [ROS type name]\n'
+    def __str__(self):
```

Comment:
While not being changed in this PR I wouldn't expect the string to have a trailing newline. If the user prints this string it would result in an empty trailing line (since `print` adds its own newline at the end).